fe/fe-core/src/main/java/com/starrocks/planner/FragmentNormalizer.java

```diff
@@ -421,11 +421,18 @@ boolean isSimpleRegionPredicate(Expr expr) {
     boolean hasNonDeterministicFunctions(Expr expr) {
         if (expr instanceof FunctionCallExpr) {
             FunctionCallExpr callExpr = (FunctionCallExpr) expr;
-            if (FunctionSet.nonDeterministicFunctions.contains(callExpr.getFn().functionName())) {
+            String funcName = callExpr.getFn().functionName();
+            if (FunctionSet.nonDeterministicFunctions.contains(funcName)) {
+                return true;
+            }
+            if (FunctionSet.NOW.equals(funcName)) {
+                return true;
+            }
+            if (FunctionSet.nonDeterministicTimeFunctions.contains(funcName) && callExpr.getChildren().isEmpty()) {
```

Comment:
What will happen if `callExpr.getChildren().isEmpty()` is false?

Reply:
For an example:

unix_timestamp("2023-12-12 13:48:31") -- has children, it is a deterministic function;
unix_timestamp() -- has no children, it is just like unix_timestamp(now()), it is a deterministic function.